utils/commands.js

```diff
@@ -0,0 +1,199 @@
+  }
+  spinner.succeed();
+
+  const dotRemakeReady = writeDotRemake(dotRemakeObj);
+
+  if (dotRemakeReady) {
+    log(boxen(getSuccessMessage(projectDir), boxenOptions));
+  }
+}
+
+const clean = () => {
+  let dotRemakeObj = readDotRemake();
+  if (!dotRemakeObj) {
+    log(chalk.bgRed('You are not in the root directory of a remake project.'));
+    process.exit();
+  }
+  spinner = ora('Cleaning project.').start();
+  shell.rm('-rf', '.cache/');
+  shell.rm('-rf', '_remake/dist');
+  spinner.succeed();
+  // TODO - replace above statement with bellow statement once the framework is 
+  // updated with the clean script
+  // shell.exec('npm run clean');
+}
+
+const build = () => {
+  let dotRemakeObj = readDotRemake();
+  if (!dotRemakeObj) {
+    log(chalk.bgRed('You are not in the root directory of a remake project.'));
+    process.exit();
+  }
+  spinner = ora('Building project.').start();
+  shell.exec('npm run build', { silent: true });
+  spinner.succeed();
+}
+
+const serve = () => {
+  log('TODO');
+}
+
+const deploy = async () => {
+  clean();
+  build();
+
+  let dotRemakeObj = readDotRemake();
+  if (!dotRemakeObj) {
+    log(chalk.bgRed('You are not in the root directory of a remake project.'));
+    process.exit();
+  }
+  await registerUser();
+
+  if (!dotRemakeObj.projectName) {
+    const subdomainAnswer = await inquirer.prompt([questions.INPUT_SUBDOMAIN]);
+    spinner = ora(`Checking if ${subdomainAnswer.subdomain}.remakeapps.com is available`).start();
+    
+    // check if name is available
+    const isSubdomainAvailable = await checkSubdomain(subdomainAnswer.subdomain);
+    if (!isSubdomainAvailable) {
+      spinner.fail(`${subdomainAnswer.subdomain}.remakeapps.com is not available`);
+      process.exit();
+    }
+    spinner.succeed(`${subdomainAnswer.subdomain}.remakeapps.com is available`);
+    
+    // prompt yes to confirm
+    const confirmSubdomainAnswer = await inquirer.prompt([questions.CONFIRM_SUBDOMAIN]);
+    if (confirmSubdomainAnswer.deployOk === false) {
+      log(chalk.bgRed('Stopped deployment.'));
+      process.exit();
+    }
+
+    spinner = ora(`Registering ${subdomainAnswer.subdomain}`).start();
+    const subdomainRegistered = await registerSubdomain(subdomainAnswer.subdomain);
+    if (!subdomainRegistered.success) {
+      spinner.fail(subdomainRegistered.message)
+      process.exit();
+    }
+    spinner.succeed(`${subdomainAnswer.subdomain}.remakeapps.com is belonging to your app.`);
+    
+    spinner = ora(`Writing .remake file.`).start();
+    dotRemakeObj.projectName = subdomainAnswer.subdomain
+    const writtenDotRemake = writeDotRemake(dotRemakeObj)
+    if (!writtenDotRemake) {
+      spinner.fail('Could not write subdomain to .remake');
+      process.exit();
+    }
+    spinner.succeed();
+  }
+
+  try {
+    await createDeploymentZip(dotRemakeObj.projectName);
+    await pushZipToServer(dotRemakeObj.projectName);
+    removeDeploymentZip(dotRemakeObj.projectName);
```

Comment:
This is so clean. I love how I can just read these 3 lines and know what they do.